### ISSUE TITLE
Adaptable for linux?

### ISSUE THREAD

**Issue by matlink** [pirate/security-growler#10]
As PushAlotAuth is quite content weak and seems abandoned, do you think this project can be adapted for linux-based systems? I can see that the notification script is made for OSX, then I think is hackable for linux.

**Comment by pirate** [comment]
Yes it certainly can be, in fact I tried to make it as cross-platform as possible.  The only mac-specific part currently is the `osxnotifiactions.py` logger.

The only major thing that needs to be done is to make a menubarlet for Linux that displays the output of `menubar.sh`, and passes click events as the text of the item clicked as an argument, e.g.
`./menubar.sh "Settings..."`.

I personally don't use desktop linux for day-to-day work, so I don't have the time to invest in making a Linux version, but feel free to fork/submit a pr!

**Comment by nmaggioni** [comment]
Integration with Gnome's notifications is possible (as per [Arch Wiki](https://wiki.archlinux.org/index.php/Desktop_notifications#Usage_in_programming) and [Gnome's specs](https://developer.gnome.org/notification-spec/), or the [relative bindings docs](https://lazka.github.io/pgi-docs/#Notify-0.7)), and in fact I've achived it with the attached patch.

The scripts starts and produces notifications, but I noticed another important problem: parsing the system's logs. For example I use `systemd`, and reading the system's logs is only possible by piping the output of the `journalctl` command; it isn't possible to just read a file as the script is currently doing.

Anyway, this patch might be a good starting point for whoever wants to work at this port.

#### [gnome_patch.txt](https://github.com/pirate/security-growler/files/244220/gnome_patch.txt)

**Comment by pirate** [comment]
Thanks for this patch!

I'd rather have the gnome notification be a seperate logger rather than overload `osxnotifications.py`.   The platform detection can actually happen inside of `settings.py`, which then just sets the proper logger list on app launch.

As for the systemd problem, it's not an issue at all, we can simply add a new `logger` source and specify the output of `popen('journalctl').readline()` as the source of new lines instead of a file.

**Comment by nmaggioni** [comment]
Great suggestion, I didn't consider it. My Python-fu is rusty...

I'll look into this and eventually begin working on a proper PR, what do you reckon?

**Comment by pirate** [comment]
Thanks!  I probably wont have time to tackle full linux support myself in the near future, but if you submit a PR for a prototype, I'd be happy to help test/refactor/bugfix it.

Always happy to answer any questions as well, shoot me a hangouts/fb invite nikisweeting@gmail.com if you want to chat rather than question/answer on github issues.
